### ISSUE TITLE
Create a simple http server: Browser request failed.

### ISSUE THREAD

**Issue by Dogs-toeight** [eclipse-vertx/vert.x#4312]
Hello! First of all, I'm not sure if this is a problem with my use or if it's a bug:
I tried to use Vertx4.2.5 and OpenJdk 17.0.2 create a simple http application.
### Code
```
public static void main(String[] args) {
        HttpServer server = Vertx.vertx(new VertxOptions()).createHttpServer();

        server.requestHandler(request -> {

            // This handler gets called for each request that arrives on the server
            HttpServerResponse response = request.response();
            response.putHeader("content-type", "text/plain");

            // Write to the response and end it
            response.end("Hello World!");
            response.close();
        });

        server.listen(8080);
    }
```
As you see, I'm using the example from the official documentation. And I request http://localhost:8080/ :
<img width="569" alt="image" src="https://user-images.githubusercontent.com/33596244/158526732-46abc73c-5180-4241-b582-2e7477322fe6.png">
No response. The browser show ERR_EMPTY_RESPONSED. 

I initially thought it was my version of Vertx that was incompatible with the JDK version, So I tried to lower the Vertx version 4.2.4.
It‘s work. 

Are there any changes to httpServer creation in 4.2.5? If yes, can you give an example? Thanks!

**Comment by vietj** [comment]
you should not close the response, that will close the connection

**Comment by Dogs-toeight** [comment]
> you should not close the response, that will close the connection

@vietj Hey!
I stopped `response.close()`, the result is the same.

**Comment by vietj** [comment]
can you use wireshark to see what goes on the wire ?

**Comment by Dogs-toeight** [comment]
> can you use wireshark to see what goes on the wire ?
> […](#)
> On Wed, Mar 16, 2022 at 8:53 AM L Jackie ***@***.***> wrote: you should not close the response, that will close the connection @vietj Hey! I stopped response.close(), the result is the same. — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you were mentioned.Message ID: ***@***.***>

@vietj Hello, I followed your suggestion to use wireshark for packet listening.
Then I saw a problem:
Run my code on a Windows machine, It's work. The packet :
<img width="1254" alt="image" src="https://user-images.githubusercontent.com/33596244/158613381-b1f72cc7-e4fa-4b16-8d17-eb90b158a776.png">
Run my code on a MacOS machine, no response. The packet: 
<img width="1791" alt="image" src="https://user-images.githubusercontent.com/33596244/158613736-24a2b6b2-ca6d-47fa-ae4b-8cef80044f75.png">

**Comment by haizz** [comment]
I have the same problem after upgrading from 3.9.9 to 4.2.6. Works with 4.2.4.

```
% curl -v http://localhost:8000/
*   Trying ::1...
* TCP_NODELAY set
* Connected to localhost (::1) port 8000 (#0)
> GET / HTTP/1.1
> Host: localhost:8000
> User-Agent: curl/7.64.1
> Accept: */*
>
* Empty reply from server
* Connection #0 to host localhost left intact
curl: (52) Empty reply from server
* Closing connection 0
```

**Comment by Dogs-toeight** [comment]
> I have the same problem after upgrading from 3.9.9 to 4.2.6.

I try to find problem. I use WareShark for packet capture analysis. I found that after establishing the TCP connection, the client sent an HTTP packet, and the server sent a 'FIN' ：
<img width="1613" alt="image" src="https://user-images.githubusercontent.com/33596244/162351917-7afb6935-5b75-42f9-86e8-b7701c07cbee.png">
And I found it worked fine on my Windows system. Running on MacOS causes this problem.
The Vertx team didn't give any advice, maybe they think there's something wrong with my operation
The funny thing is that I used the official code.
Finally, I solution was to upgrade to 4.2.4.
